src/ripple/protocol/impl/STValidation.cpp

```diff
@@ -28,33 +28,50 @@ namespace ripple {
 
 STValidation::STValidation(
     uint256 const& ledgerHash,
+    std::uint32_t ledgerIndex,
     uint256 const& consensusHash,
     NetClock::time_point signTime,
     PublicKey const& publicKey,
+    SecretKey const& secretKey,
     NodeID const& nodeID,
-    bool isFull)
+    bool isFull,
+    FeeSettings const& fees,
+    std::vector<uint256> const& amendments)
     : STObject(getFormat(), sfValidation), mNodeID(nodeID), mSeen(signTime)
 {
-    // Does not sign
+    assert (mNodeID.isNonZero ());
     setFieldH256 (sfLedgerHash, ledgerHash);
     setFieldH256 (sfConsensusHash, consensusHash);
     setFieldU32 (sfSigningTime, signTime.time_since_epoch().count());
 
     setFieldVL (sfSigningPubKey, publicKey.slice());
-    assert (mNodeID.isNonZero ());
-
     if (isFull)
         setFlag (kFullFlag);
-}
 
-uint256 STValidation::sign (SecretKey const& secretKey)
-{
+    setFieldU32(sfLedgerSequence, ledgerIndex);
```

Comment:
There is that sequence vs index. I'd personally go with sequence, but that is just IMHO.